magpie-data/src/main/java/io/openraven/magpie/data/utils/EntityTypeResolver.java

```diff
@@ -80,13 +81,16 @@ public void init(JavaType javaType) {
     }
 
     @Override
-    public String idFromValue(Object o) {
-        return idFromValueAndType(o, o.getClass());
+    public String idFromValue(Object object) {
+      return idFromValueAndType(object, object.getClass());
     }
 
     @Override
-    public String idFromValueAndType(Object o, Class<?> aClass) {
-        return null;
+    public String idFromValueAndType(Object object, Class<?> aClass) {
+      if (object instanceof Resource) {
+        return ((Resource) object).getResourceType();
+      }
+      return aClass.getName();
```

Comment:
Previously under Jackson version 2.13.5 by returning null here and `include = JsonTypeInfo.As.PROPERTY` we were getting following: 

```
{"resourceType":"null",
"resourceType":"AWS::RDS::DBInstance",
"documentId":null,
"arn":null,"resourceName":null,"resourceId":null,"awsRegion":null,"awsAccountId":null,"createdIso":null,"updatedIso":null,"discoverySessionId":null,"maxSizeInBytes":null,"sizeInBytes":null,"configuration":null,"supplementaryConfiguration":null,"tags":null,"discoveryMeta":null}
```

Now Jackson 2.15.4 doesn't allow to set null to resolving id.
With above fix we are getting now:
```
{"resourceType":"AWS::RDS::DBInstance",
"resourceType":"AWS::RDS::DBInstance",
"documentId":null,"arn":null,
"resourceName":null,"resourceId":null,"awsRegion":null,"awsAccountId":null,"createdIso":null,"updatedIso":null,"discoverySessionId":null,"maxSizeInBytes":null,"sizeInBytes":null,"configuration":null,"supplementaryConfiguration":null,"tags":null,"discoveryMeta":null}
```

When we set `include = JsonTypeInfo.As.EXISTING_PROPERTY` we are getting:
```
{"resourceType":"AWS::RDS::DBInstance",
"documentId":null,"arn":null,
"resourceName":null,"resourceId":null,"awsRegion":null,"awsAccountId":null,"createdIso":null,"updatedIso":null,"discoverySessionId":null,"maxSizeInBytes":null,"sizeInBytes":null,"configuration":null,"supplementaryConfiguration":null,"tags":null,"discoveryMeta":null}
```